src/dev.ts

```diff
@@ -58,13 +58,16 @@ export async function loadDev(id: string, options: ResolvedVitePWAOptions, viteC
         // the sw precache (self.__SW_MANIFEST) will be empty since we're using `dev-dist` folder
         // we only need to add the navigateFallback if configured
         const navigateFallback = options.workbox.navigateFallback
+        // we need to exclude the manifest.webmanifest from the sw precache: avoid writing it to the dev-dist folder
+        const webManifestUrl = options.devOptions.webManifestUrl ?? `${options.base}${options.manifestFilename}`
         const { filePaths } = await generateServiceWorker(
           Object.assign(
             {},
             options,
             {
               workbox: {
                 ...options.workbox,
+                navigateFallbackDenylist: [new RegExp(`^${webManifestUrl}$`)],
```

Comment:
Do we need to escape regex here?

Reply:
no, using the constructor will replace with the escaped literal: `/manifest.webmanifest` => `^\/manifest.webmanifest$`.

I check it.

Reply:
wait, I can double test...

Reply:
It is ok:

![imagen](https://user-images.githubusercontent.com/6311119/172438082-5ade0489-f595-45dd-971d-dcd480844254.png)